docs/secret-rotation.md

```diff
@@ -0,0 +1,254 @@
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <MAKE SURE THIS MATCHES “cloudCredentialsSecretName:” IN configmap.yaml>
+  namespace: capi-self
+  annotations:
+    # Allow the sealed secret controller to take over this secret after bootstrapping
+    sealedsecrets.bitnami.com/managed: "true"
+stringData:
+  clouds.yaml: |
+    clouds:
+      openstack:
+        auth:
+          auth_url: ...
+          application_credential_id: ...
+          application_credential_secret: ...
+        interface: ...
+        identity_api_version: 3
+        auth_type: ...
+```
+
+From here make sure that the terminal’s current working directory is in the same one as **credentials.yaml**, `fluxcd-config/clusters/<cluster-name>/`, and then run the following Kubeseal command:
+
+| kubeseal \--kubeconfig kubeconfig \--format yaml \--controller-name sealed-secrets \--controller-namespace sealed-secrets-system \--secret-file credentials.yaml  \--sealed-secret-file encrypted-creds.yaml |
+| :---- |
+
+| Note that this will create a new file named 'encrypted-creds.yaml' containing the encrypted contents and will need to be renamed to 'credentials.yaml'. This is just to be safe, however this step can be avoided by replacing 'encrypted-creds.yaml' with 'credentials.yaml' in the above command. |
+| :---- |
+
+<a id="updating-secrets-link"></a>
+### **Updating Secrets**
+
+With the new secrets encrypted it is time to change the sealed secret which is on the kubernetes cluster.
+
+<!-- The image on the right is the first sealed secret which is going to replaced. This is being viewed using the **k9s** UI. -->
+
+The process of changing the secret over is as simple as adding, committing and pushing the new **credentials.yaml** to the upstream GitHub repository which is being tracked by FluxCD.
+
+| It is highly recommended to `git switch -c` to a new branch when pushing the new credentials.yaml so that a pull request can be reviewed before merging to the tracked branch. |
+| :---- |
+
+When the new **credentials.yaml** has been merged into the FluxCD tracked repository branch, FluxCD will try to update the **sealed secret** on the self managed cluster once it recognises that there are differences between the cluster’s current configuration and the one on GitHub.
+
+| The time interval in which FluxCD will check for these *drifts between configurations* is set in the cluster’s **helmrelease.yaml** found within the `fluxcd-config/components/<cluster-name>/` directory, under the `interval` variable. |
+| :---- |
+
+This should result in the **sealed secret** on the kubernetes cluster to contain the new secret.
+
+<a id="deleting-old-secrets-link"></a>
+###	**Deleting Old Secrets**
+
+Once the sealed secret has been updated on kubernetes it is time to do some *housekeeping* by now deleting the old credentials and unnecessary files created along the way, including, but not limited to, **old-credentials.yaml** and its corresponding application credential on Horizon.
```

Comment:
Why 'not limited to'? Let's try to be precise about what does and doesn't need deleted please 🙂

Reply:
Reworded.